### ISSUE TITLE
Dependency not found: @firebase/app-exp in ./node_modules/@firebase/database/dist/index.esm.js

### ISSUE THREAD

**Issue by stratpoint-ced** [firebase/firebase-js-sdk#4779]
### Describe your environment

  * Operating System version: macOS Mojave 10.14.6
  * Browser version: Chrome 89.0.4389.114
  * Firebase SDK version: 8.4.0
  * Nodejs Version: 14.16.0
  * NPM version: 7.6.3
  * Firebase Product: database

### Describe the problem
 Updating the firebase npm package from 8.1.1 to 8.4.0 throws an error when running the development build.
#### Steps to reproduce:
1. I use the command ```npm i firebase@8.4.0 --save``` to update the target version. After the installation, this gives a warning like this.
<img width="507" alt="Screen Shot 2021-04-13 at 2 44 47 PM" src="https://user-images.githubusercontent.com/74229036/114509998-be68c080-9c68-11eb-802d-919ec4f7bef4.png">

2. Then when running development build command throws an error:
<img width="459" alt="Screen Shot 2021-04-13 at 2 34 17 PM" src="https://user-images.githubusercontent.com/74229036/114507529-d7bc3d80-9c65-11eb-9113-ea1cf663e170.png">

I already tried to delete node_modules and run ```npm cache clean --force``` but none of these works. Still the error encountered.

**Comment by DanPurdy** [comment]
Seeing this too in my ci pipeline with the dependabot update from 1.2.7 to 1.2.8

**Comment by humphd** [comment]
This started happening to us on GitHub CI last night as well (we can't reproduce locally), and it seems to affect most PRs now:

```
 Cannot find module '@firebase/app-exp' from 'src/api/users/node_modules/firebase/node_modules/@firebase/database/dist/index.node.cjs.js'

    Require stack:
      src/api/users/node_modules/firebase/node_modules/@firebase/database/dist/index.node.cjs.js
      src/api/users/node_modules/firebase/dist/index.node.cjs.js
      src/api/users/node_modules/@firebase/rules-unit-testing/dist/index.cjs.js
      src/api/users/test/e2e/users.test.js

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:306:11)
      at Object.<anonymous> (src/api/users/node_modules/firebase/node_modules/@firebase/database/dist/index.node.cjs.js:8:14)
```

**Comment by humphd** [comment]
In the past hour, this has stopped happening in our CI builds (re-triggering failed builds from last night).  I'm not sure if it's "fixed" but it doesn't happen every time anymore for me.

**Comment by looptheloop88** [comment]
Hi @stratpoint-ced, thanks for bringing this matter to our attention. I was able to replicate this issue when trying to update Firebase from an older version to 8.4.0. Per checking, this seems to be fixed in version 8.4.1. Please try to update and let us know if the issue persists.

**Comment by stratpoint-ced** [comment]
Thank you @looptheloop88 for the quick response. It is working smoothly now.
